consumer/consumererror/component.go

```diff
@@ -0,0 +1,38 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package consumererror // import "go.opentelemetry.io/collector/consumer/consumererror"
+
+import "go.opentelemetry.io/collector/component"
+
+// Component contains information about the component that returned an
+// error. This is helpful when reasoning about the context an error
+// occurred in when handling errors in upstream components within a
+// pipeline.
+type Component struct {
+	err error
+	id  component.ID
+}
+
+var _ error = Component{}
+
+// NewComponent instantiates a new Component error.
+// `id` should be the component's component.ID object.
+func NewComponent(err error, id component.ID) error {
```

Comment:
I didn't add options here to prevent us from taking on dead code once `consumer` is stable. If we have ideas for what sorts of data could go in here, I will add an options parameter.